src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -259,6 +260,18 @@ public void apply(Project project) {
             minecraftClassesArtifact = createArtifacts.map(task -> project.files(task.getCompiledArtifact()));
         }
 
+        var supplyDevLogin = project.provider(() -> extension.getRuns().stream().anyMatch(model -> model.getDevLogin().getOrElse(false)));
+
+        // Create a configuration to resolve DevLogin
+        var devLoginConfig = project.getConfigurations().create("devLoginConfig", spec -> {
+            spec.setDescription("This configuration is used to inject DevLogin into the runtime classpath of modding source sets");
+            // We only add DevLogin as a default dependency to allow people to overwrite it
```

Comment:
Just add it all the time. People can upgrade it easily by adding it, or downgrade it with a version resolution rule.